connectors/services/job_execution.py

```diff
@@ -121,20 +72,18 @@ async def _sync(self, sync_job):
                 )
                 return
 
-        job_type = sync_job.job_type
-
-        if job_type in [JobType.FULL, JobType.INCREMENTAL]:
-            await self._content_sync(sync_job, connector, source_klass)
-
-        elif (
-            job_type == JobType.ACCESS_CONTROL
-            and connector.features.document_level_security_enabled()
-        ):
-            await self._access_control_sync(sync_job, connector, source_klass)
+        if not self.should_execute(connector, sync_job):
+            return
 
-        else:
-            sync_job.log_error(
-                f"Unsupported job type '{job_type}'. Skipping sync job execution..."
+        sync_job_runner = SyncJobRunner(
+            source_klass=source_klass,
+            sync_job=sync_job,
+            connector=connector,
+            es_config=self._override_es_config(connector),
+        )
+        if not self.sync_job_pool.try_put(sync_job_runner.execute):
+            sync_job.log_info(
+                f"{self.display_name.capitalize()} service is already running {self.max_concurrency} sync jobs and can't run more at this poinit. It will retry in {self.idling} seconds. Increase '{self.max_concurrency_config}' in config if you want the service to run more sync jobs."  # pyright: ignore
```

Comment:
This has potential as being spammy (e.g. every 30 seconds by default it'll apper) - should it be debug?

It's also too long IMO, can we shorten it? I think the part with `self.idling` can go.